Fix #2207 - make feature position instances pickable

macraild opened the pull request:
This adds \_\_getnewargs\_\_ to SeqFeatures.*Positions classes where required, so that instances are (un)picklable

This pull request addresses issue #2207 

<!--- Please read each of the following items and confirm by replacing
 !--the [ ] with a [X] --->

- [X ] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X ] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [ ] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

peterjc commented on Bio/SeqFeature.py:
```diff
@@ -1695,6 +1695,22 @@ class WithinPosition(int, AbstractPosition):
     >>> p2 == 13
     True
 
+
```
Could you move this to the ``__getnewargs__`` docstring instead?

peterjc commented on Bio/SeqFeature.py:
```diff
@@ -1804,6 +1824,22 @@ class BetweenPosition(int, AbstractPosition):
 
     i.e. For equality (and sorting) the position objects behave like
     integers.
+
```
Again, could you move this to the ``__getnewargs__`` docstring please.

peterjc commented on Bio/SeqFeature.py:
```diff
@@ -2039,6 +2079,20 @@ class OneOfPosition(int, AbstractPosition):
     >>> p2 == 1901
     True
 
+    To allow pickling and unpickling of class instances, the arguments
```
Same comment, this isn't of general interest but ``__getnewargs__`` lacks any docstring at all - this would work better there I think.

peterjc commented:
I'd like a cosmetic change in the docstring placement, but otherwise the change looks good.

Are you comfortable adding something to ``Tests/test_SeqFeature.py`` or do you think the doctests are enough?

macraild commented:
I agree, the docstrings were out of place, so I've rewritten as method-level docstrings. But doctests don't seem to run from here, so I've converted the original doctests  to unittests in Test/SeqFeature.py. Note that the scope of the tests is unchanged. To the extent I understand what is required here, these tests cover it, but happy to add more if there is something specific I have missed.

peterjc commented:
Doctests are fragile, so having explicit unittests seems better all round. Are you OK with a squash-merge? Do you want your name in the ``NEWS.rst`` and ``CONTRIB.rst``?

macraild commented:
Yes, squash-merge is fine. I'm happy to have my name in NEWS and CONTRIB if you think it is warranted.

peterjc commented:
Thank you, applied.
